routes/user.js

```diff
@@ -0,0 +1,296 @@
+const Mail = require("@sendgrid/mail");
+
+cloudinary.config({
+  cloud_name: process.env.CLOUD_API_NAME,
+  api_key: process.env.CLOUD_API_KEY,
+  api_secret: process.env.CLOUD_API_SECRET,
+});
+const storage = multer.diskStorage({
+  destination: "../upload/images",
+  filename: (req, file, callback) => {
+    return callback(
+      null,
+      `${file.fieldname}_${Date.now()}${path.extname(file.originalname)}`
+    );
+  },
+});
+
+const upload = multer({
+  storage: storage,
+});
+
+Mail.setApiKey(process.env.SENDGRID_API_KEY);
+
+function generateAccessToken(username) {
+  return jwt.sign(username, process.env.TOKEN_SECRET);
+}
+
+app.post("/register", async (req, res) => {
+  const salt = await bcrypt.genSalt();
+  const userPassword = await bcrypt.hash(req.body.password, salt);
+
+  if (req.body.password !== req.body.confirmpassword) {
+    res.status(500).json({
+      error: 1,
+      message: "password not matched",
+      data: null,
+    });
+  } else {
+    const user_post = new user({
+      username: req.body.username,
+      firstname: req.body.firstname,
+      lastname: req.body.lastname,
+      password: userPassword,
+      email: req.body.email,
+    });
+    user_post.save(function (err, row) {
+      if (err) {
+        res.status(500).json({
+          error: 1,
+          message: err.message,
+          data: null,
+        });
+      } else {
+        const msg = {
+          to: `${req.body.email}`, // Change to your recipient
+          from: "ankit@excellencetechnologies.info", // Change to your verified sender
+          subject: "Sending with SendGrid is Fun",
+          text: "and easy to do anywhere, even with Node.js",
+          html: `<strong>${req.body.username}  has been registered</strong>`,
+        };
+        Mail.send(msg)
+          .then(() => {
+            console.log("Email sent");
+          })
+          .catch((error) => {
+            console.error(error);
+          });
+        res.status(200).json({
+          error: 0,
+          message: "saved successfully",
+          data: username,
+        });
+      }
+    });
+  }
+});
+
+app.post("/login", passport.authenticate("local"), function (req, res) {
+  var username = req.user.name;
+  const token = generateAccessToken({ username: req.body.username });
+  res.status(200).json({
+    error: 0,
+    message: "user list",
+    data: token,
+  });
+});
+
+app.get("/user/get", jwtAuth, async function (req, res) {
+  user
+    .findOne({ _id: req.user._id })
+    .populate("userdata")
+    .then((user) => {
+      res.status(200).json({
+        error: 0,
+        message: "user list",
+        data: userData,
+      });
+    });
+});
+
+app.put("/delete", jwtAuth, async function (req, res) {
+  var user_id = req.user.user_id;
+  await user.deleteOne({ _id: user_id }, function (err, results) {
+    if (err) {
+      res.status(501).json({
+        error: 1,
+        message: "user not found",
+        data: null,
+      });
```

Comment:
after successfully delete of user, delete user address.
await and call back should not use at same time